### PR TITLE
Add support for app_state metricset

### PR DESCRIPTION
**Pull request by narph**
## What does this PR do?

Adds support for light  app_state metricset.

## Why is it important?
metricbeat-azure-app-state-overview

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

**Comment by andresrc** [comment]
Related: https://github.com/elastic/beats/issues/19881

File: x-pack/metricbeat/module/azure/_meta/config.yml [hunk 1]
```diff
@@ -111,3 +111,11 @@
 #  api_key: ''
 #  metrics:
 #   - id: ["requests/count", "requests/duration"]
+
+#- module: azure
+#  metricsets:
+#    - app_state
+#  enabled: true
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/module/azure/_meta/config.yml:118]: Is `enabled: true` required here?

Comment (narph) [reply x-pack/metricbeat/module/azure/_meta/config.yml:118]: not necessary , just making sure users are aware of this flag

File: metricbeat/docs/modules/azure/app_state.asciidoc [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+[[metricbeat-metricset-azure-app_state]]
+[role="xpack"]
+=== Azure app_state metricset
+
+beta[]
+
+include::../../../../x-pack/metricbeat/module/azure/app_state/_meta/docs.asciidoc[]
+
+This is a default metricset. If the host module is unconfigured, this metricset is enabled by default.
```
Comment (kaiyan-sheng) [inline metricbeat/docs/modules/azure/app_state.asciidoc:13]: Whats host module here?

File: x-pack/metricbeat/module/azure/module.yml [hunk 3]
```diff
@@ -4,5 +4,6 @@ metricsets:
  - container_instance
  - container_service
  - database_account
+ - app_state
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/module/azure/module.yml:7]: Nit: Maybe rewrite this list in alphabetical order?

File: metricbeat/docs/modules/azure/app_state.asciidoc (hunk 2 above)
Comment (narph) [reply metricbeat/docs/modules/azure/app_state.asciidoc]: fixed, it was set as default in the manifest file

**Review by kaiyan-sheng — APPROVED** [review]
lgtm!

**Pull request merged**